### ISSUE TITLE
Quantum Espresso GeoOpt within ASE 

### ISSUE THREAD

**Issue by daisiduke** [libAtoms/workflow#124]
Hi all,
coming back to our discussion [here](https://github.com/libAtoms/ExPyRe/issues/22) about geometry optimization with Quantum Espresso:
I incorporated the QE GeoOpt via the PreconLBFGS ASE solver by basically merging the respective SPC approach (wfl.calculators.espresso) and the already existing GeoOpt with a given ML potential (wfl.generate.optimize).
Here is my code suggestion [relax.zip](https://github.com/libAtoms/workflow/files/9091583/relax.zip).
Other than just adding the corresponding espresso_relax.py to the calculators two only minor changes within the wfl.calculators.dft file are necessary
```
from wfl.calculators import castep, vasp, espresso, espresso_relax
[...]
    elif calculator_name == "QE_GO":
        op = espresso_relax.evaluate_autopara_wrappable
[...]
```
However, after looking through the code again and giving it some thought I realized that this approach is somewhat redundant since the it is also possible to call the wfl.generate.optimize module with an appropriate QE calculator. Preparing this calculator is somewhat lengthy though and also already incorporated in the wfl.calculators.espresso module. 
Maybe the most elegant approach would be to split calculator preparation and calculation call. 
I tried to sketch this in the following figure. What do you think?
![image](https://user-images.githubusercontent.com/22477583/178496468-39009c9d-a5e5-4ef7-8e48-743c5898244b.png)

**Comment by bernstei** [comment]
I agree that given that evaluate, optimize, and MD all use a calculator, it makes sense to unify the way they do it. I guess that's most like your "design with two modules", although I'm not sure I like the "prepare" as an input to eval/optim/MD, as opposed to a module with utility functions that all 3 can call (since things like that need to be pickled, and an object that contains a QUIP Potential can't be pickled).

@gelzinyte did we have any ideas about unifying `calculators.generic` and `calculators.dft`?

**Comment by bernstei** [comment]
@gelzinyte For now I'll take another look at the ASE-extended ORCA calculator, and see if I have any comments relevant to doing the something similar to the other DFT calculators. If there isn't one yet, I'll open an issue for the DFT calculator update, and refer to it here.

**Comment by daisiduke** [comment]
Thanks for the quick responses! In general it does not hurry since I for now just generate the requiered QE calculator within my script. It just appeared useful and I guess "utility function" was the term I was looking for. I just still have to learn :) If there is any way I can contribute, let me know. Looking forward to the updates
